superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/index.jsx

```diff
@@ -50,7 +50,7 @@ const propTypes = {
   columns: PropTypes.arrayOf(columnType),
   savedMetricsOptions: PropTypes.arrayOf(savedMetricType),
   savedMetric: savedMetricType,
-  datasourceType: PropTypes.string,
+  datasource: PropTypes.object.isRequired,
```

Comment:
if datasource is marked as `isRequired` then you shouldn't need the `datasource?.type` below and can just do `datasource.type`. But if datasource isn't passed in sometimes, then you should remove the `isRequired` part